### DISCUSSION TITLE
Debugger deploy and launch error after selecting one of the options from the dialog box

### DISCUSSION THREAD

**Discussion by mahi9115** [nextgenhealthcare/connect#5140] [General]
Mirth Connect Version - 4.0.0

I was trying out the new Debugging function recently and changed the headless config on mcserver.vmoptions and mcservices.vmoptions to false.

The debugger does not launch after I select one of the options in the JS debugger dialog box and throws an error on the console as following.

![image](https://user-images.githubusercontent.com/98981373/163601257-805fa88f-adf5-46d6-9e5b-9ba53f5c82b6.png)

![image](https://user-images.githubusercontent.com/98981373/163601281-e63e29da-4d4f-4967-84d6-45aef2ade325.png)

I have also tried setting the headless options in one of the file to be true an vice-versa on both the file, and also left both the setting to true sill getting the same error. 

Is there anything specific I need to check to make it work?

**Comment by ajadams2** [comment]
I get the same error.  I am using OpenJDK 13.

**Comment by ChristopherSchultz** [comment]
Looks like a bug to me. The server shouldn't be trying to initialize any GUI stuff.

**Reply by tonygermano** [reply]
This changed in mirth 4.0. The new integration with the Rhino debugger requires it to run in the same JVM as the server. This feature is optional, of course, and not recommended to run on production servers.

**Comment by matthydehoog** [comment]
Any solution yet??

**Reply by pacmano1** [reply]
For those that posted nothing about what operating system  and/or java release and/or how you installed Mirth, it's hard to help you.  

Next time you post, it would be a good idea to add
Window 10 Enterprise
OpenJDK 13
Mirth 4.0.1 installed using the x64 windows installer, all defaults accepted.

For Windows, try:  Stop mirth in windows services.  Start mirth via "mcserver", i.e. the one that starts mirth in a cmd prompt and try again.  Windows services cannot interact with the desktop by default.

**Comment by rutledj** [comment]
Will the debugger work when launching MC from the MC Administrator Launcher? Our Mirth run on Mirth appliances on VM machines. How can you run it in this situation?

**Reply by pacmano1** [reply]
Gotta be local (i.e. on the desktop) of the mirth server for it to work.

**Comment by matthydehoog** [comment]
[Enable Debugger in WIndows.pdf](https://github.com/nextgenhealthcare/connect/files/8668208/Enable.Debugger.in.WIndows.pdf)
I have added the steps to the NextGen description

**Comment by ajadams2** [comment]
On Windows 10 and have mirth installed on my laptop. Using OpenJDK 13 and PostgreSQL 13.
I upgraded to 4.01. I was previously using 4.0. I changed the values from true to false in both files listed in the pdf, mcservice.vmoptions and mcsever.vmoptions. Stopped the service via task manager -> services.  I manually ran mcserver.exe as administrator and it now works. 

I plan on testing if making changes to only 1 of the files makes a difference or not.  Thanks for the write up too.

**Comment by cinexin** [comment]
Is there a way to debug scripts locally with the server running on Docker?
